File: web-service/src/main/java/obby0511/sentiment/analysis/Application.java
package obby0511.sentiment.analysis;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.http.MediaType;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.RouterFunctions;
import org.springframework.web.reactive.function.server.ServerResponse;

import static org.springframework.web.reactive.function.server.RequestPredicates.POST;
import static org.springframework.web.reactive.function.server.RequestPredicates.accept;

@SpringBootApplication
public class Application {

    @Bean
    public RouterFunction<ServerResponse> route(Handler handler) {
        return RouterFunctions.route(
                POST("/").and(accept(MediaType.TEXT_PLAIN)), handler::handle);
    }

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
}

File: web-service/src/main/kotlin/obby0511/sentiment/analysis/Application.kt
package obby0511.sentiment.analysis

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import org.springframework.http.MediaType.APPLICATION_JSON
import org.springframework.web.reactive.function.server.router

@SpringBootApplication
class Application {

    @Bean
    fun route(handler: Handler) = router {
        accept(APPLICATION_JSON).nest {
            POST("/", handler::analyze)
        }
    }
}

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}
